src/routes/pull.rs

```diff
@@ -27,3 +45,31 @@ fn pull() -> Result<Json<Vec<Manifest>>, Error> {
     }
     Ok(Json(manifests))
 }
+
+#[get("/search?<filter>")]
+fn search_filter(filter: Option<ManifestFilter>) -> Result<Json<Vec<Manifest>>, Error> {
+    let mut manifests = Vec::new();
+    for path in glob::glob(&format!("{}/**/*.toml", *RAVEN_REPOSITORY_PATH))? {
+        let path = path?;
+        let mut file = File::open(path)?;
+
+        // Allocate a string long enough to hold the entire file
+        let mut s = file
+            .metadata()
+            .map(|m| String::with_capacity(m.len() as usize))
+            .unwrap_or_default();
+
+        file.read_to_string(&mut s)?;
+        manifests.push(toml::from_str(&s)?);
+    }
+
+    if let Some(p) = filter {
```

Comment:
`p` is not a good name. It's not even the abbreviation of `ManifestFilter`. 

(Same for `n` and `c` later)